qualtran/surface_code/algorithm_summary.py

```diff
@@ -0,0 +1,43 @@
+#  Copyright 2023 Google LLC
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#      https://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+
+
+from attrs import field, frozen
+
+_PRETTY_FLOAT = field(default=0.0, repr=lambda x: f'{x:g}')
+
+
+@frozen
+class AlgorithmSummary:
```

Comment:
https://github.com/quantumlib/Qualtran/pull/356/files#r1304942596

is this class redundant with `magic_state_factory.MagicStateCount`? Can you refactor to remove code duplication?

You responded  

> is about the T state factory not the algorithm specs, right?

I don't know what this means (?) `MagicStateCount` is used to describe the number of magic states required by an algorithm

Reply:
removed MagicStateCount

Reply:
but this means either using toffoli instead of ccz count or creating another field

Reply:
toffoli and ccz only differ by some hadamards, so I think we should lump them together.